### PR TITLE
IBP-3833 Study Germplasm APIs

### PR DESCRIPTION
**Pull request by abatac**

### CODE REVIEW

**Comment by abatac** [comment]
Hi @clarysabel FYI I added a new api to create study germplasm based on a specified germplasm list. Also for adding germplasm descriptor to the study, I'm thinking, instead of creating a separate api to add germplasm descriptor to the study, maybe we can reuse this existing api (**/{crop}/programs/{programUUID}/studies/{studyId}/datasets/{datasetId}/variables**) since currently we just store these germplasm descriptors at plot dataset's projectprop. We would just need to adjust this api to allow germplasm descriptor variables. What do you think?

File: src/main/java/org/ibp/api/rest/study/StudyGermplasmResource.java [hunk 1]
```diff
@@ -7,26 +7,46 @@
...
+		@PathVariable final String programUUID,
+		@PathVariable final Integer studyId, @PathVariable final Integer entryId, @RequestBody final StudyGermplasmDto studyGermplasmDto) {
+		return new ResponseEntity<>(this.studyGermplasmService.replaceStudyGermplasm(studyId, entryId, studyGermplasmDto),
+			HttpStatus.OK);
+
+	}
+
+	@ApiOperation(value = "Create germplasm entries in study based on the specified germplasm list",
+		notes = "Create germplasm entries in study based on the specified germplasm list")
+	@RequestMapping(value = "/{cropname}/programs/{programUUID}/studies/{studyId}/germplasm/list/{germplasmListId}", method = RequestMethod.POST)
```
Comment (clarysabel) [inline src/main/java/org/ibp/api/rest/study/StudyGermplasmResource.java:41]: Hi @abatac 
I think the service should be named entry-lists and germplasmList should be included in the body

File: src/main/java/org/ibp/api/rest/study/StudyGermplasmResource.java [hunk 2]
```diff
@@ -7,26 +7,46 @@
...
-            notes = "Replace germplasm entry in study")
-    @RequestMapping(value = "/{cropname}/programs/{programUUID}/studies/{studyId}/germplasm/{entryId}", method = RequestMethod.PUT)
-    @ResponseBody
-    public ResponseEntity<StudyGermplasmDto> replaceStudyGermplasm(final @PathVariable String cropname, @PathVariable final String programUUID,
-                                                                   @PathVariable final Integer studyId, @PathVariable final Integer entryId, @RequestBody final StudyGermplasmDto studyGermplasmDto) {
-        return new ResponseEntity<>(this.studyGermplasmService.replaceStudyGermplasm(studyId, entryId, studyGermplasmDto),
-                HttpStatus.OK);
+	@ApiOperation(value = "Replace germplasm entry in study",
+		notes = "Replace germplasm entry in study")
+	@RequestMapping(value = "/{cropname}/programs/{programUUID}/studies/{studyId}/germplasm/{entryId}", method = RequestMethod.PUT)
```
Comment (clarysabel) [inline src/main/java/org/ibp/api/rest/study/StudyGermplasmResource.java:29]: @abatac germplasm should be renamed by entries

File: src/main/java/org/ibp/api/rest/study/StudyGermplasmResource.java [hunk 3]
```diff
@@ -7,26 +7,46 @@
...
+		return new ResponseEntity<>(this.studyGermplasmService.replaceStudyGermplasm(studyId, entryId, studyGermplasmDto),
+			HttpStatus.OK);
+
+	}
+
+	@ApiOperation(value = "Create germplasm entries in study based on the specified germplasm list",
+		notes = "Create germplasm entries in study based on the specified germplasm list")
+	@RequestMapping(value = "/{cropname}/programs/{programUUID}/studies/{studyId}/germplasm/list/{germplasmListId}", method = RequestMethod.POST)
+	@ResponseBody
+	public ResponseEntity<List<StudyGermplasmDto>> createStudyGermplasm(final @PathVariable String cropname,
```
Comment (clarysabel) [inline src/main/java/org/ibp/api/rest/study/StudyGermplasmResource.java:43]: FYI I am creting a new DTO called StudyEntryDto. 
After some conversations with Mariano, he shared the ideas about how germplasm descriptors will work. So I need to make this Dto more dynamic. I hope to have something working soon. Our work will be overlaped. Let's discuss with @darla-leafnode if it makes sense to continue working on this ticket in this sprint

**Pull request closed without merging**